iot-hub/Samples/device/ImportExportDevicesSample/ProcessImpExpCommands.cs

```diff
@@ -0,0 +1,438 @@
+
+        // Name of blob container holding the work data.
+        public static string containerName = "devicefiles";
+
+        // Name of the file used for exports and imports. 
+        // This is set by the IoT SDK, and can't be changed.
+        public static string deviceListFile = "devices.txt";
+
+        // List of devices to add. This is a temporary file.
+        public static string devices_to_add = "devices_new.txt";
+
+        /// <summary>
+        /// Sets up references to the blob hierarchy objects, sets URI_to_container with an SAS for access.
+        /// Create the container if it doesn't exist.
+        /// </summary>
+        /// <returns></returns>
+        public static string PrepareForImportExport()
+        {
+            // Get reference to storage account.
+            // This is the storage account used to hold the import and export files.
+            CloudStorageAccount cloudStorageAccount =
+                 CloudStorageAccount.Parse(storageAccountConnectionString);
+
+            // Get reference to the blob client.
+            CloudBlobClient cloudBlobClient = cloudStorageAccount.CreateCloudBlobClient();
+
+            // Get reference to the container to be used.
+            cloudBlobContainer = cloudBlobClient.GetContainerReference(containerName);
+
+            // Get the URI to the container. This doesn't have an SAS token (yet).
+            string containerURI = cloudBlobContainer.Uri.ToString();
+
+            // How to get reference to a blob. 
+            //   Just leaving this in here in case you want to copy this block of code and use it for this.
+            // CloudBlockBlob cloudBlockBlob = cloudBlobContainer.GetBlockBlobReference(deviceListFile);
+            // This is how you get the URI to the blob. The import and export both use a container
+            //   with container-level access, so this isn't used for importing or exporting the devices.
+            // Just leaving it in here for when it's needed.
+            // string blobURI = blockBlob.Uri.ToString();
+
+            try
+            {
+                // The call below will fail if the sample is configured to use the storage emulator 
+                //   in the connection string but the emulator is not running.
+                // Change the retry policy for this call so that if it fails, it fails quickly.
+                BlobRequestOptions requestOptions = new BlobRequestOptions() { RetryPolicy = new NoRetry() };
+                cloudBlobContainer.CreateIfNotExistsAsync(requestOptions, null);
+            }
+            catch (StorageException)
+            {
+                throw;
+            }
+
+            // Call to get the SAS token for container-level access.
+            string containerSASToken = GetContainerSasToken(cloudBlobContainer);
+
+            // Append the SAS token to the URI to the container. This is returned.
+            string URI_to_container = containerURI + containerSASToken;
```

Comment:
NIT: consistent naming

Reply:
renamed URI_to_container to containerURI.